hazelcast-client-new/src/test/java/com/hazelcast/client/listeners/ListenerLeakTest.java

```diff
@@ -0,0 +1,224 @@
+    public void testMultiMapEntryListeners() {
+        testListenerLeak(new IFunction() {
+            @Override
+            public Object apply(Object input) {
+                HazelcastInstance client = (HazelcastInstance) input;
+                MultiMap multiMap = client.getMultiMap(randomString());
+                String id = multiMap.addEntryListener(new EntryAdapter(), false);
+                assertTrue(multiMap.removeEntryListener(id));
+                return id;
+            }
+        });
+    }
+
+    @Test
+    public void testListListeners() {
+        testListenerLeak(new IFunction() {
+            @Override
+            public Object apply(Object input) {
+                HazelcastInstance client = (HazelcastInstance) input;
+                IList<Object> list = client.getList(randomString());
+                String id = list.addItemListener(new ItemListenerAdapter(), false);
+                assertTrue(list.removeItemListener(id));
+                return id;
+            }
+        });
+    }
+
+    @Test
+    public void testSetListeners() {
+        testListenerLeak(new IFunction() {
+            @Override
+            public Object apply(Object input) {
+                HazelcastInstance client = (HazelcastInstance) input;
+                ISet<Object> set = client.getSet(randomString());
+                String id = set.addItemListener(new ItemListenerAdapter(), false);
+                assertTrue(set.removeItemListener(id));
+                return id;
+            }
+        });
+    }
+
+    @Test
+    public void testQueueListeners() {
+        testListenerLeak(new IFunction() {
+            @Override
+            public Object apply(Object input) {
+                HazelcastInstance client = (HazelcastInstance) input;
+                IQueue<Object> queue = client.getQueue(randomString());
+                String id = queue.addItemListener(new ItemListenerAdapter(), false);
+                assertTrue(queue.removeItemListener(id));
+                return id;
+            }
+        });
+    }
+
+    @Test
+    public void testReplicatedMapListeners() {
+        testListenerLeak(new IFunction() {
+            @Override
+            public Object apply(Object input) {
+                HazelcastInstance client = (HazelcastInstance) input;
+                ReplicatedMap<Object, Object> replicatedMap = client.getReplicatedMap(randomString());
+                String id = replicatedMap.addEntryListener(new EntryAdapter<Object, Object>());
+                assertTrue(replicatedMap.removeEntryListener(id));
+                return id;
+            }
+        });
+    }
+
+    @Test
+    public void testDistributedObjectListeners() {
+        testListenerLeak(new IFunction() {
+            @Override
+            public Object apply(Object input) {
+                HazelcastInstance client = (HazelcastInstance) input;
+                String id = client.addDistributedObjectListener(new DistributedObjectListener() {
```

Comment:
I would use a mock here since you don't care about its content. 

String id = client.addDistributedObjectListener(mock(DistributedObjectListener.class));

This increased the readability of the code.